### ISSUE TITLE
align.rotation_matrix works if a and b array have different shape

### ISSUE THREAD

**Issue by kain88-de** [MDAnalysis/mdanalysis#1152]
### Expected behaviour

a ValueError should be raised

### Actual behaviour

works just fine

### Code to reproduce the behaviour

``` python
import MDAnalysis as mda
from MDAnalysis.analysis.align import rotation_matrix

u = mda.Universe(top, trj)
rotation_matrix(u.atoms.positions, u.atoms.positions[:10]

```

### Currently version of MDAnalysis:
(run `python -c "import MDAnalysis as mda; print(mda.__version__)"`)
dev
